### ISSUE TITLE
[Bug]: Panic occurred at runtime. Message:  slice index starts at 1 but ends at 0

### ISSUE THREAD

**Issue by Clarkkkk** [web-infra-dev/rspack#5535]
### Version

```sh
System:
    OS: Windows 11 10.0.22621
    CPU: (16) x64 Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
    Memory: 19.67 GB / 47.89 GB
  Browsers:
    Edge: Chromium (121.0.2277.83)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @rsbuild/core: ^0.3.10 => 0.3.10
```

### Details

It may have something to do with `require.context` and the config `resolve.extensions`. It only occurs in rsbuild, and rspack seems ok.

### Reproduce link

https://github.com/Clarkkkk/rspack-dynamic-import-issue/tree/rsbuild_issue

### Reproduce Steps

1. `pnpm i`
2. `npm run dev:rsbuild`

**Comment by chenjiahan** [comment]
Error message:

![image](https://github.com/web-infra-dev/rspack/assets/7237365/47430032-0797-4d98-a471-6ab4aa2422fb)

**Comment by Clarkkkk** [comment]
Hi, sorry to bother but is there any chance that this fix can be released before Chinese new year?

**Comment by chenjiahan** [comment]
Yeah we plan to release a new version this week (before Chinese new year)

**Comment by Clarkkkk** [comment]
Awesome! Thank you all for all the work you've done!
